### PR TITLE
Add fips-static version of kairos packages

### PR DESCRIPTION
**Pull request by mudler**
So we can use them in all the flavors

### CODE REVIEW

File: packages/system/immucore/collection.yaml [hunk 1]
```diff
@@ -21,3 +21,15 @@ packages:
       - https://github.com/kairos-io/immucore
     license: "Apache License v2"
     description: "The Kairos immutability management interface"
+  - name: "immucore"
+    category: "fips-static"
```
Comment (Itxaka) [inline packages/system/immucore/collection.yaml:25]: should fips static just be fips? I mean, does it help to have 2 different fips binaries when we could have one that should work everywhere?

File: packages/system/immucore/collection.yaml [hunk 2]
```diff
@@ -21,3 +21,15 @@ packages:
...
+    ldflags: "-linkmode external -extldflags -static"
```
Comment (Itxaka) [inline packages/system/immucore/collection.yaml:26]: missing ldflags for the non-static fips version

File: packages/system/immucore/collection.yaml (hunk 1 above)
Comment (mudler) [reply packages/system/immucore/collection.yaml:25]: I don't know if it has implication in runtime yet, hence why a new category

File: packages/system/immucore/collection.yaml (hunk 2 above)
Comment (mudler) [reply packages/system/immucore/collection.yaml:26]: no need, using `with`

**Comment by mudler** [comment]
mmm locally builds fine:

```
 INFO   # command-line-arguments
       /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/libcrypto.a(libcrypto-lib-dso_dlfcn.o): in function `dlfcn_globallookup':
       (.text+0x11): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /tmp/go-link-3916787737/000003.o: in function `mygetgrouplist':
       /_/os/user/getgrouplist_unix.go:15: warning: Using 'getgrouplist' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /tmp/go-link-3916787737/000002.o: in function `mygetgrgid_r':
       /_/os/user/cgo_lookup_unix.go:37: warning: Using 'getgrgid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /tmp/go-link-3916787737/000002.o: in function `mygetgrnam_r':
       /_/os/user/cgo_lookup_unix.go:42: warning: Using 'getgrnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /tmp/go-link-3916787737/000002.o: in function `mygetpwnam_r':
       /_/os/user/cgo_lookup_unix.go:32: warning: Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /tmp/go-link-3916787737/000002.o: in function `mygetpwuid_r':
       /_/os/user/cgo_lookup_unix.go:27: warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /tmp/go-link-3916787737/000008.o: in function `_cgo_2ac87069779a_C2func_getaddrinfo':
       /tmp/go-build/cgo_unix.cgo2.c:58: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/libcrypto.a(libcrypto-lib-bio_sock.o): in function `BIO_gethostbyname':
       (.text+0x81): warning: Using 'gethostbyname' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
       
 INFO   Removing intermediate container 422142b6d46e
        ---> 6288f346fb51
       Step 14/14 : RUN chmod +x /system/discovery/kcrypt-discovery-challenger
 INFO    ---> Running in 7ebe36deffdb
 INFO   Removing intermediate container 7ebe36deffdb
        ---> 8db0241fe030
       Successfully built 8db0241fe030
 INFO   Successfully tagged luet/cache:afafa54fea81203cb7fa00e407ae284d026b1c1b7166cf9f3bf68b72dac69741
 INFO   SUCCESS 🐋  Building image luet/cache:afafa54fea81203cb7fa00e407ae284d026b1c1b7166cf9f3bf68b72dac69741 done
 INFO   📦  fips-static/kcrypt-challenger-0.5.0 🔨  Generating delta
```

**Comment by mudler** [comment]
PEBCAK. I forgot to commit the fix - was working locally but not remotely :facepalm:

**Comment by mudler** [comment]
yesterday while trying to fix the error I've added https://github.com/kairos-io/packages/pull/324/commits/2498af384a45fcbfdf1a7112d18e1ec7b7950870 locally but forgot to commit it. from that point I forgot to have it there and assumed wasn't solving the issue.. instead it was the needed fix :sweat_smile:

**Review by mauromorales — APPROVED** [review]
lgtm

**Pull request merged**